src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryTaskLauncher.java

```diff
@@ -70,7 +70,9 @@ public TaskStatus status(String id) {
 
 	protected Mono<TaskStatus> toTaskStatus(Throwable throwable, String id) {
 		if ((throwable instanceof CloudFoundryException)
-			&& ((CloudFoundryException) throwable).getCode() == 10010) {
+			&& (Integer.valueOf(10010).equals(((CloudFoundryException) throwable).getCode())
```

Comment:
This is testing whether we're dealing with an http 404 not found error. I think it got more complicated because of the two versions of the API.

I will extract bits in more descriptive variables for now, not sure what you guys can do at the client level